### PR TITLE
Heap Sort in Java

### PR DESCRIPTION
**Pull request by ritvij14**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
Yes
(Write your answer here.)

### Description
This is the Heap Sort algorithm in Java that sorts a binary heap in ascending order. Time complexity is O(nLogn).
(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
Issue #648 
(Write your answer here.)

### CODE REVIEW

**Comment by ritvij14** [comment]
@siddharth25pandey @Kajol-Kumari @atarax665 @iamrajiv @plazzy99 please review this.

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@ritvij14 Almost same as [this](https://www.geeksforgeeks.org/java-program-for-heap-sort/) only at some places variable name are changed.
So, to avoid this do the following modifications
* Create a swap function for line 47-49 & 18-20
* Calculate the time consumed during each execution & display it.

**Comment by ritvij14** [comment]
@siddharth25pandey please have a look, I did the changes you asked, also calculating the time at multiple places using `System.nanoTime()`.

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@ritvij14 Looks fine to me
Able to merge.
Thanks!

**Review by siddharth25pandey — APPROVED** [review]

**Review by Kajol-Kumari — APPROVED** [review]
Codewise LGTM!

File: Java/sort/HeapSort.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+Time to make first max heap:
+1.0418000000000001E-5
+Time to make sort using max heaps and putting largest element in the end:
+7.484000000000001E-6
+Time to sort the array:
+0.002423689
+Array after heap sort:
+1 3 6 7 9 10 11 12 20
+
+Time complexity = O(nLogn)
```
Comment (iamrajiv) [inline Java/sort/HeapSort.java]: @ritvij14 please add more description about time complexity like time for heapify and building the heap, what you have provide is the overall time complexity. Also, add space complexity.

**Review by iamrajiv — CHANGES_REQUESTED** [review]

**Review by iamrajiv — APPROVED** [review]
@ritvij14 LGTM!

**Pull request merged**